prometheus/prometheus_test.go

```diff
@@ -54,6 +54,64 @@ func TestNewPrometheusSink(t *testing.T) {
 		t.Fatalf("Unregister(sink) = false, want true")
 	}
 }
+// TestMultiplePrometheusSink tests registering multiple sinks on the same registerer with different descriptors
+func TestMultiplePrometheusSink(t *testing.T) {
+	gaugeDef := GaugeDefinition{
+		Name: []string{"my", "test", "gauge"},
+		Help: "A gauge for testing? How helpful!",
+	}
+
+	cfg := PrometheusOpts{
+		Expiration:         5 * time.Second,
+		GaugeDefinitions:   append([]GaugeDefinition{}, gaugeDef),
+		SummaryDefinitions: append([]SummaryDefinition{}),
+		CounterDefinitions: append([]CounterDefinition{}),
+		Name: "sink1",
+	}
+
+	sink1, err := NewPrometheusSinkFrom(cfg)
+
+	reg := prometheus.DefaultRegisterer
+
+	if reg == nil {
+		t.Fatalf("Expected default register to be non nil, got nil.")
+	}
+
+	if err != nil {
+		t.Fatalf("err = %v, want nil", err)
+	}
```

Comment:
Minor style recommendation: generally errors should be checked immediately after the line the where they are assigned, with no extra newline between the statements. That formatting is encoded in https://github.com/mvdan/gofumpt as "No empty lines before a simple error check".



```suggestion
	sink1, err := NewPrometheusSinkFrom(cfg)
	if err != nil {
		t.Fatalf("err = %v, want nil", err)
	}
	
	reg := prometheus.DefaultRegisterer
	if reg == nil {
		t.Fatalf("Expected default register to be non nil, got nil.")
	}
```